frontend/src/main/components/UCSBDates/UCSBDatesTable.js

```diff
@@ -46,18 +43,22 @@ export default function UCSBDatesTable({ dates, currentUser }) {
         }
     ];
 
-    if (hasRole(currentUser, "ROLE_ADMIN")) {
-        columns.push(ButtonColumn("Edit", "primary", editCallback, "UCSBDatesTable"));
-        columns.push(ButtonColumn("Delete", "danger", deleteCallback, "UCSBDatesTable"));
-    } 
+    const columnsIfAdmin = [
+        ...columns,
+        ButtonColumn("Edit", "primary", editCallback, "UCSBDatesTable"),
+        ButtonColumn("Delete", "danger", deleteCallback, "UCSBDatesTable")
+    ]
+
+    const columnsToDisplay = () => {
+        return hasRole(currentUser, "ROLE_ADMIN") ? columnsIfAdmin : columns;
+    }
```

Comment:
Why are you wrapping this value in a function if you are just going to call it immediately? Can't you just replace this with 
```suggestion
    const columnsToDisplay = hasRole(currentUser, "ROLE_ADMIN") ? columnsIfAdmin : columns;
```

Reply:
Good question. I implemented your change just now.